### ISSUE TITLE
Custom language syntax highlighting seems not to work

### ISSUE THREAD

**Issue by mac501pl** [react-monaco-editor/react-monaco-editor#312]
**Describe the bug**
After defining a custom language, syntax highlighting only works for the first line of code. I tried reproducing the issue on [Monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-custom-languages) but it works as expected there.

**To Reproduce**
App.jsx:
```
import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
import * as React from "react";
import MonacoEditor from "react-monaco-editor";
import { ReactElement } from "react";

export default class App extends React.Component {
	constructor(props: Record<string, unknown>) {
		super(props);
	}
	editorWillMount(monacoEditor: typeof monaco): void {
		monacoEditor.languages.register({ id: "8080asm" });
		monacoEditor.languages.setMonarchTokensProvider("8080asm", {
			tokenizer: {
				root: [
					[ /([\w]+:)/, "label" ]
				]
			}
		});
		monacoEditor.editor.defineTheme("8080-dark", {
			base: "vs",
			inherit: false,
			rules: [
				{ token: "label", foreground: "ff0000" }
			],
			colors: {}
		});
	}
	render(): ReactElement {
		return (
			<MonacoEditor
				height="800"
				theme="8080-dark"
				language="8080asm"
				editorWillMount={this.editorWillMount.bind(this)}
			/>
		);
	}
}
```

Webpack config:
```
const { merge } = require("webpack-merge");
const CopyPkgJsonPlugin = require("copy-pkg-json-webpack-plugin");
const HtmlWebpackPlugin = require("html-webpack-plugin");
const MiniCssExtractPlugin = require("mini-css-extract-plugin");
const OptimizeCssAssetsWebpackPlugin = require("optimize-css-assets-webpack-plugin");
const TerserPlugin = require("terser-webpack-plugin");
const MonacoWebpackPlugin = require("monaco-editor-webpack-plugin");
const webpack = require("webpack");
const path = require("path");

function srcPaths(src) {
	return path.resolve(__dirname, src);
}

const isEnvProduction = process.env.NODE_ENV === "production";
const isEnvDevelopment = process.env.NODE_ENV === "development";

// #region Common settings
const commonConfig = {
	devtool: isEnvDevelopment ? "source-map" : false,
	mode: isEnvProduction ? "production" : "development",
	output: { path: srcPaths("dist") },
	node: { __dirname: false, __filename: false },
	resolve: {
		alias: {
			"@": srcPaths("src"),
			"@main": srcPaths("src/main"),
			"@models": srcPaths("src/models"),
			"@public": srcPaths("public"),
			"@renderer": srcPaths("src/renderer"),
			"@utils": srcPaths("src/utils"),
			"@assets": srcPaths("assets")
		},
		extensions: [".js", ".json", ".ts", ".tsx"]
	},
	optimization: {
		minimize: true,
		minimizer: [
			new OptimizeCssAssetsWebpackPlugin()
			// new TerserPlugin()
		]
	},
	module: {
		rules: [
			{
				test: /\.(ts|tsx)$/,
				exclude: /node_modules/,
				loader: "ts-loader"
			},
			{
				test: /\.scss$/,
				exclude: /node_modules/,
				use: [MiniCssExtractPlugin.loader, "css-loader", "sass-loader"]
			},
			{
				test: /\.css$/,
				include: /node_modules/,
				use: ["style-loader", "css-loader"]
			},
			{ test: /\.json$/, loader: "json", include: "/assets/" },
			{
				test: /\.(jpg|png|svg|ico|icns)$/,
				loader: "file-loader",
				options: {
					name: "[path][name].[ext]"
				}
			},
			{
				test: /\.ttf$/,
				use: ["file-loader"]
			}
		]
	}
};
// #endregion
const mainConfig = merge(commonConfig, {
	entry: "./src/main/main.ts",
	target: "electron-main",
	output: {
		filename: "main.bundle.js"
	},
	plugins: [
		new CopyPkgJsonPlugin({
			remove: ["scripts", "devDependencies", "build"],
			replace: {
				main: "./main.bundle.js",
				scripts: { start: "electron ./main.bundle.js" },
				postinstall: "electron-builder install-app-deps"
			}
		})
	]
});
const rendererConfig = merge(commonConfig, {
	entry: "./src/renderer/renderer.tsx",
	target: "electron-renderer",
	output: {
		filename: "renderer.bundle.js"
	},
	plugins: [
		new HtmlWebpackPlugin({
			template: path.resolve(__dirname, "./public/index.html")
		}),
		new MiniCssExtractPlugin({
			filename: "[name].css"
		}),
		new MonacoWebpackPlugin()
		// new webpack.optimize.LimitChunkCountPlugin({
		// 	maxChunks: 1
		// })
	]
});
module.exports = [mainConfig, rendererConfig];
```

**Expected behavior**
Syntax highlighting works for the whole file

**Environment (please complete the following information):**
 - OS: Windows
 - Browser Chromium (Electron)
 - Bundler webpack

**Comment by domoritz** [comment]
This looks like a question about Monaco. This repo is for the React wrapper for Monaco. Please file issues about Monaco in the Monaco repo. If you find that you cannot reproduce the issue with just Monaco, I can reopen this issue.

**Comment by mac501pl** [comment]
Sorry if asked my question in a wrong place, I did verify the issue both on Monaco Editor Playground and by pulling in its standalone version into my project. Both work as expected.

During my experiments I also noticed that syntax highlights correctly after it get scrolled off screen (e.g. when interacting with the minimap), which I found a bit odd.

If you're interested the repo in question is: https://github.com/mac501pl/8BitSimulator
